### PR TITLE
Fix NTP set race condition

### PR DESCRIPTION
**Pull request by Pavithrab7**
There's currently a problem with phosphor-timesyncd, where enabling NTP doesn't immediately reflect in the system status on return[1].  To say it another way, NTP is not enabled/disabled atomically, which leads to the following problem.

// Disable NTP
PATCH /redfish/v1/Managers/bmc/NetworkProtocol
{"NTP":{"ProtocolEnabled": false}}
// Set the time manually
PATCH /redfish/v1/Managers/bmc {"DateTime": "<timestring"}

Doing this in rapid succession leads to a 500 error, which is obviously a bug.  In the prior commit, this error was changed to a PropertyValueConflict error, which is still incorrect, but at least informative of what's going on.  REST APIs are intended to have CRUD compliance.  The response should not be returned until the value has been accepted, and not doing that can lead to problems.

This commit changes the backend to use systemd directly, rather than routing through phosphor-settings, to avoid this race.

Quite possibly resolves #264 but haven't tested that.

Tested: The above procedure succeeds.

[1] https://github.com/systemd/systemd/pull/11424

Change-Id: Id53af9bf05bec052dd8013ae10c3c5e55f033038
Signed-off-by: Pavithra Barithaya <pavithrabarithaya07@gmail.com>

### CODE REVIEW

**Comment by gtmills** [comment]
add to approvelist

**Comment by Pavithrab7** [comment]
yes that's the upstream commit.

**Review by gtmills — APPROVED** [review]

File: redfish-core/lib/network_protocol.hpp [hunk 1]
```diff
@@ -238,30 +238,29 @@ inline void getNetworkData(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
...
-    if (ntpEnabled)
-    {
-        timeSyncMethod = "xyz.openbmc_project.Time.Synchronization.Method.NTP";
-    }
-    else
+    if (ec)
     {
-        timeSyncMethod =
-            "xyz.openbmc_project.Time.Synchronization.Method.Manual";
+        BMCWEB_LOG_DEBUG("Failed to set elapsed time. DBUS response error {}",
```
Comment (baemyung) [inline redfish-core/lib/network_protocol.hpp:246]: Would this is `BMCWEB_LOG_ERROR()`?

Comment (Pavithrab7) [reply redfish-core/lib/network_protocol.hpp:246]: `BMCWEB_LOG_DEBUG() `would also be fine? - https://gerrit.openbmc.org/c/openbmc/bmcweb/+/69948/23/redfish-core/lib/network_protocol.hpp#246 ?

Comment (baemyung) [reply redfish-core/lib/network_protocol.hpp:246]: @Pavithrab7 
I think - it needs to be updated to ERROR for 1110 and also for upstream, as it generates `internalError()`.

Comment (Pavithrab7) [reply redfish-core/lib/network_protocol.hpp]: updated to ERROR for 1110.

**Pull request merged**